### DISCUSSION TITLE
Deleting Package Data

### DISCUSSION THREAD

**Discussion by cmurray219** [modflowpy/flopy#1683] [General]
I'm unable to find a function that will remove/delete/overwrite an existing package in MF6.  Can someone point me to the right function?

I am importing an existing model using:

`sim=flopy.mf6.MFSimulation.load()`

I successfully made some changes to hydraulic conductivity in the NPF package by simply creating a new instance of the package as:

```
npf = flopy.mf6.modflow.mfgwfnpf.ModflowGwfnpf(
    gwf, 
    icelltype=1,
    k=hk1,
    k33=vk1,
    save_flows=True
)
```
I get a warning that "Package with type npf already exists. Replacing existing package." which is exactly what I want.

The problem is that I also want to adjust some specified heads in the CHDpackage.  I do the following:

```
chd_array = chd.stress_period_data.get_data(key=0)
new_chd_array = []
for item in chd_array:
    if item[1] == 434.0:
        one_row = (item[0], 430.0)
        new_chd_array.append(one_row)
    elif item[1] == 420.0:
        one_row = (item[0], 415.0)
        new_chd_array.append(one_row)
        
chd = flopy.mf6.ModflowGwfchd(
    gwf,
    stress_period_data=new_chd_array, 
    maxbound = len(new_chd_array),
    save_flows = True
)
```
It correctly creates the new CHD package, but it also keeps the existing one.  (I guess having multiple CHD packages is allowed, unlike having multiple NPF packages.)  So there are two CHD packages.  When I run the code again, there are 3, and so on.  I'm going to put this in a loop and run a whole array of head and hydraulic conductivity sets, so I can't have it adding new CHD packages every time.  I need either a function to delete the original CHD package after I've copied out the information I need or a setting/parameter to force it to overwrite like the NPF package.

Any ideas?

**Comment by jlarsen-usgs** [comment]
@cmurray219 

you can use the `remove_package()` method to remove a package from a model. Here's an example:

```python
sim = flopy.mf6.MFSimulation.load()
gwf = sim.get_model("my_model_name")
gwf.remove_package("CHD")
```

**Reply by cmurray219** [reply]
Thank you!
